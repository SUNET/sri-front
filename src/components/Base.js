import React from "react";
import { Route, Switch } from "react-router-dom";

import SearchContainer from "../containers/Search";
import { ContactDetails } from "./Contact";

class Base extends React.Component {
    render() {
        return (
            <main className="container">
                <Switch>
                    <Route path="/contacts" component={SearchContainer} />
                    <Route
                        path="/contact/:contactId"
                        component={ContactDetails}
                    />
                    <Route path="/roles" component={SearchContainer} />
                </Switch>
            </main>
        );
    }
}

export default Base;
