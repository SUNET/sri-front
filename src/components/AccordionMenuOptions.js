import React from "react";
import { Nav } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import ToggleSection, { ToggleHeading, TogglePanel } from "../components/ToggleSection";

import "../style/AccordionMenuOptions.scss";

class AccordionMenuOptions extends React.Component {
    renderHeader(option, index) {
        return (
            <ToggleHeading key={`${index}`}>
                <div className="accordion-menu-options__header">{option.header}</div>
            </ToggleHeading>
        );
    }
    renderCollapseList(subOptions, optionIndex) {
        return (
            <TogglePanel>
                {subOptions.map((subOption, subOpIndex) => {
                    return (
                        <div key={`option_menu-${optionIndex}-${subOpIndex}`} className="accordion-menu-options__link">
                            <Nav.Link
                                as={NavLink}
                                activeClassName="active"
                                to={subOption.route}
                                onClick={() => this.props.onSelectedOption()}
                            >
                                <i className={`accordion-menu-options__link__icon ${subOption.iconClass}`}></i>
                                <span className="accordion-menu-options__link__name">{subOption.name}</span>
                            </Nav.Link>
                        </div>
                    );
                })}
            </TogglePanel>
        );
    }
    render() {
        const { data } = this.props;

        return (
            <div className="accordion-menu-options">
                {data.map((option, optionIndex) => {
                    return (
                        <ToggleSection key={`option_menu-${optionIndex}`}>
                            {this.renderHeader(option, optionIndex)}
                            {this.renderCollapseList(option.subOptions, optionIndex)}
                        </ToggleSection>
                    );
                })}
            </div>
        );
    }
}

export default AccordionMenuOptions;
