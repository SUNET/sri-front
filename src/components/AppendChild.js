import React from 'react';
import { Field } from 'redux-form';
import { withTranslation } from 'react-i18next';
import FieldInput from './FieldInput';

class AppendChild extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      fields: [{ name: '' }],
    };
  }

  componentWillMount() {}

  handleFieldNameChange = (index) => (event) => {
    const newShareholders = this.state.fields.map((field, sindex) => {
      if (index !== sindex) return field;
      return { ...field, name: event.target.value };
    });

    this.setState({ fields: newShareholders });
  };

  handleAddFild = (event) => {
    event.preventDefault();
    this.setState({
      fields: this.state.fields.concat([{ name: '' }]),
    });
  };

  render() {
    let { t } = this.props;
    return (
      <div>
        {this.state.fields.map((field, index) => (
          <Field
            component={FieldInput}
            type="text"
            placeholder={`${field.name}${index + 1}`}
            value={field.name}
            onChange={this.handleFieldNameChange(index)}
          />
        ))}
        <button className="btn btn-add outline" onClick={(e) => this.handleAddField(e)}>
          <span>{t('actions/add-new')}</span>
        </button>
      </div>
    );
  }
}

export default withTranslation()(AppendChild);
