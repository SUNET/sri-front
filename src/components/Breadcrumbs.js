import React from 'react';
import Breadcrumb from 'react-bootstrap/Breadcrumb';
import PropTypes from 'prop-types';
import Routes from '../Routes';

function generatePathsArray(pathString, isDetailsEntity, entityDetails) {
  return pathString.split('/').map((pathElement, index, array) => {
    const mathIdCurrentDetailsInLocationPath = pathString.includes(entityDetails.id);
    const isLastPath = index === array.length - 1;
    const isActive = isLastPath;
    let pathTo = `/${pathElement}`;
    let name = Routes[pathTo];

    if (typeof Routes[pathTo] === 'undefined') {
      pathTo = array.slice(0, index + 1).join('/');
      name = isLastPath && isDetailsEntity && mathIdCurrentDetailsInLocationPath ? entityDetails.name : Routes[pathTo];
    }
    return {
      pathTo,
      name,
      isActive,
    };
  });
}

const Breadcrumbs = ({ history, isDetailsEntity, entityDetails }) => {
  const { pathname } = history.location;
  const routeSteps = generatePathsArray(pathname, isDetailsEntity, entityDetails);
  return (
    <div className="breadcrumbs">
      <Breadcrumb>
        {routeSteps.map(({ pathTo, name, isActive }, index) => {
          return (
            <Breadcrumb.Item active={isActive} key={`${index * Math.random()}`} href={pathTo}>
              {name}
            </Breadcrumb.Item>
          );
        })}
      </Breadcrumb>
    </div>
  );
};

Breadcrumbs.propsTypes = {
  history: PropTypes.object.isRequired,
  isDetailsEntity: PropTypes.bool.isRequired,
  entityDetails: PropTypes.bool.isRequired,
};

export default Breadcrumbs;
