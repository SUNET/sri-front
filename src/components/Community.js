import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { withTranslation } from "react-i18next";

import SearchOrganizationContainer from "../containers/organization/SearchOrganization";
import SearchContactContainer from "../containers/contact/SearchContact";
import SearchGroupContainer from "../containers/group/SearchGroup";
// import SubMenu from "./SubMenu";
// import SubMenuActions from "./SubMenuActions";

import "../style/Footer.scss";

class Community extends React.Component {
    render() {
        return (
            <Switch>
                <Redirect exact from="/community" to="/community/organizations" />
                <Route path="/community/organizations" component={SearchOrganizationContainer} />
                <Route path="/community/contacts" component={SearchContactContainer} />
                <Route path="/community/groups" component={SearchGroupContainer} />
            </Switch>
        );
    }
}

export default withTranslation()(Community);
