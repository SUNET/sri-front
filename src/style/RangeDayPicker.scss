@import "../variables.scss";

.InputFromTo {
    display: inline-block;

    label {
        margin-right: 8px;
    }

    .DayPickerInput {
        font-family: Montserrat;
        color: $secondary-color;

        &:after {
            display: inline-block;
            content: '\f073';
            font-family: "Font Awesome 5 Free";
            font-weight: 900;
            color: $complementary-color;
            position: relative;
            right: 28px;
        }

        input {
            width: 110px;
            border-radius: 16px;
            border: solid 2px $secondary-color-20;
            height: 32px;
            padding: 0 10px;
            font-size: 14px;
            &::placeholder{
                letter-spacing: -1.5px;
            }
        }

        .DayPickerInput-OverlayWrapper {
            position: absolute;
            z-index: 999;
            margin-top: 8px;

            .DayPicker-Months {
                flex-wrap: nowrap;

                .DayPicker-Caption {
                    div {
                        font-family: karla;
                        font-weight: bold;
                        font-size: 14px;
                    }
                }
            }
        }
    }

    .InputFromTo-to .DayPickerInput-Overlay {
        position: relative;
        right: 0;
    }

    .DayPicker-Day--selected:not(.DayPicker-Day--start):not(.DayPicker-Day--end):not(.DayPicker-Day--outside) {
        color: #fff;
    }

    .DayPicker-Day--selected:not(.DayPicker-Day--disabled):not(.DayPicker-Day--outside) {
        background-color: $complementary-color;
    }

    .DayPicker-Day {
        border-radius: 0 !important;
    }

    .DayPicker-Day--start {
        border-top-left-radius: 50% !important;
        border-bottom-left-radius: 50% !important;
    }

    .DayPicker-Day--end {
        border-top-right-radius: 50% !important;
        border-bottom-right-radius: 50% !important;
    }

    .DayPickerInput-Overlay {
        position: relative;
    }

    .DayPicker-NavButton--prev {
        color: $complementary-color;
    }
}
