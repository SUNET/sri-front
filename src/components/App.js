import React, { Component } from "react";
import PropTypes from "prop-types";
import { ConnectedRouter } from "connected-react-router";
import Breadcrumbs from "react-router-dynamic-breadcrumbs";
import { Route, Switch } from "react-router-dom";

import FetchingContext from "../components/FetchingContext";
import SplashContainer from "../containers/Splash";
import NotifyContainer from "../containers/Notify";
import TopHeaderContainer from "../containers/TopHeader";
import BaseContainer from "../containers/Base";
import FooterContainer from "../containers/Footer";

import { Row, Col } from "react-bootstrap";
import SideNavNetwork from "./SideNavNetwork";
import SideNavCommunity from "./SideNavCommunity";
import SubMenuActions from "./SubMenuActions";

import Routes from "../Routes";

import "bootstrap/scss/bootstrap.scss";
import "../style/App.scss";
import "../style/Breadcrumbs.scss";
import "../style/SRIButton.scss";

import { history } from "../store";

class App extends Component {
    render() {
        return (
            <FetchingContext.Provider value={this.props.is_fetching}>
                <div className="App container">
                    <ConnectedRouter history={history}>
                        <Row>
                            <Col className="px-0">
                                <TopHeaderContainer />
                            </Col>
                        </Row>
                        <Row>
                            <SplashContainer />
                            <Switch>
                                <Route path="/network" component={SideNavNetwork} />
                                <Route path="/community" component={SideNavCommunity} />
                            </Switch>
                            <Col sm={10} className="fixed-adaptative">
                                <NotifyContainer />
                                <Row className="mt-4">
                                    <Col>
                                        <Breadcrumbs mappedRoutes={Routes} />
                                    </Col>
                                    <Col className="text-right">
                                        <SubMenuActions />
                                    </Col>
                                </Row>
                                {this.props.is_app_loaded && <BaseContainer />}
                            </Col>
                        </Row>
                        <Row>
                            <Col className="px-0">
                                <FooterContainer />
                            </Col>
                        </Row>
                    </ConnectedRouter>
                </div>
            </FetchingContext.Provider>
        );
    }
}

App.propTypes = {
    is_fetching: PropTypes.bool,
    is_app_loaded: PropTypes.bool,
};

export default App;
