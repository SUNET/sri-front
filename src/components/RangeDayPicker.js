import React from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import moment from "moment";
import DayPickerInput from "react-day-picker/DayPickerInput";
import MomentLocaleUtils, { formatDate, parseDate } from "react-day-picker/moment";

import "moment/locale/en-gb";
import "moment/locale/es";
import "moment/locale/sv";

import "react-day-picker/lib/style.css";
import "../style/RangeDayPicker.scss";

export class RangeDayPicker extends React.Component {
    // static propTypes = {
    //     resetDate: PropTypes.func.isRequired
    // };

    constructor(props) {
        super(props);
        this.handleResetClick = this.handleResetClick.bind(this);
        this.from = null; // elements reference
        this.to = null; // elements reference
        this.state = {
            from: undefined,
            to: undefined
        };
    }

    showFromMonth() {
        const { from, to } = this.state;
        if (!from) {
            return;
        }
        if (moment(to).diff(moment(from), "months") < 2) {
            console.log('from: ', from);
            this.to.getDayPicker().showMonth(from);
        }
    }

    handleFromChange = (from, modifiers, dayPickerInput) => {
        // Change the from date and focus the "to" input field
        this.setState({ from }, this.props.dateFrom(from));
    };

    handleToChange = (to, modifiers, dayPickerInput) => {
        this.setState({ to }, this.showFromMonth, this.props.dateTo(to));
    };

    handleResetClick = () => {
        this.setState({ from: undefined, to: undefined }, this.props.resetDate(undefined, undefined));
        // this fixed the encapsulated input uncontrolled bug
        this.from.setState({ value: "", typedValue: "" });
        this.to.setState({ value: "", typedValue: "" });
    };

    renderDayPicker(fromOrToInput) {
        const { from, to } = this.state;
        const { t } = this.props;
        const modifiers = { start: from, end: to };
        const isFrom = fromOrToInput === "from";
        let datePickerConfig = {};
        if (isFrom) {
            datePickerConfig = {
                disabledDays: { after: to },
                toMonth: to,
                onDayClick: () => this.to.getInput().focus()
            };
        } else {
            datePickerConfig = {
                disabledDays: { before: from },
                month: from,
                fromMonth: from
            };
        }
        return (
            <div className={`input-from-to input-from-to--${fromOrToInput}`}>
                <label>{t(`filter.daypicker.${fromOrToInput}`)}</label>
                <DayPickerInput
                    value={this.state[fromOrToInput]}
                    ref={(el) => (this[fromOrToInput] = el)}
                    placeholder="dd/mm/yy"
                    format="DD/MM/YY"
                    formatDate={formatDate}
                    parseDate={parseDate}
                    dayPickerProps={{
                        ...datePickerConfig,
                        locale: "en",
                        localeUtils: MomentLocaleUtils,
                        selectedDays: [from, { from, to }],
                        modifiers,
                        numberOfMonths: 2
                    }}
                    onDayChange={isFrom ? this.handleFromChange : this.handleToChange}
                />
            </div>
        );
    }
    renderResetDatesCTA() {
        const { t } = this.props;
        return (
            <button type="button" onClick={this.handleResetClick} data-action="reset" className="range-day-picker__reset_dates btn outline btn-remove">
                {t("filter.date.clear")}
            </button>
        );
    }
    render() {
        return (
            <div className="range-day-picker">
                {this.renderDayPicker("from")}
                {this.renderDayPicker("to")}
                {this.renderResetDatesCTA()}
            </div>
        );
    }
}

export default withTranslation()(RangeDayPicker);
