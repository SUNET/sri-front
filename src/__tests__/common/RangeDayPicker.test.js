import React from 'react';
import { mount, shallow } from 'enzyme';
import { RangeDayPicker } from '../../components/RangeDayPicker';
import toJson from 'enzyme-to-json';

let RangeDayPickerWrapper;
let RangeDayPickerComponent;
let RangeDayPickerComponentInstance;

const defaultProps = {
  dateTo: jest.fn(),
  dateFrom: jest.fn(),
  resetDate: jest.fn(),
  t: jest.fn((t) => t),
};

const CreateRangeDayPickerWrapper = (props) => {
  const allProps = { ...defaultProps, ...props };
  return (
    <div>
      <RangeDayPicker {...allProps} />
    </div>
  );
};

const mountRangeDayPickerWrapper = (props) => mount(<CreateRangeDayPickerWrapper {...props} />);

const getDatePickerElements = (className) => {
  return {
    wrapper: RangeDayPickerComponent.find(className),
    label: RangeDayPickerComponent.find(className).find('label'),
    rangeDayPicker: RangeDayPickerComponent.find(className).find('DayPickerInput'),
  };
};
beforeEach(() => {
  jest.clearAllMocks();
  RangeDayPickerWrapper = mountRangeDayPickerWrapper();
  RangeDayPickerComponent = RangeDayPickerWrapper.find('RangeDayPicker');
  RangeDayPickerComponentInstance = RangeDayPickerComponent.instance();
  RangeDayPickerComponentInstance.showFromMonth = jest.fn();
});
afterEach(() => {
  RangeDayPickerWrapper.unmount();
});

describe('RangeDayPicker Component', () => {
  it('Renders correctly', () => {
    expect(toJson(RangeDayPickerWrapper)).toMatchSnapshot();
    expect(RangeDayPickerComponent.find('.range-day-picker')).toHaveLength(1);
  });
});

describe('Elements', () => {
  it('DateRange for From date', () => {
    const pickerElements = getDatePickerElements('.input-from-to--from');
    expect(pickerElements.wrapper).toHaveLength(1);
    expect(pickerElements.label).toHaveLength(1);
    expect(pickerElements.label.text()).toBe('filter.daypicker.from');
    expect(pickerElements.rangeDayPicker).toHaveLength(1);
  });
  it('DateRange for To date', () => {
    const pickerElements = getDatePickerElements('.input-from-to--to');
    expect(pickerElements.wrapper).toHaveLength(1);
    expect(pickerElements.label).toHaveLength(1);
    expect(pickerElements.label.text()).toBe('filter.daypicker.to');
    expect(pickerElements.rangeDayPicker).toHaveLength(1);
  });
  it('Reset Dates Button', () => {
    const resetButton = RangeDayPickerComponent.find('[data-action="reset"]');
    expect(resetButton.type()).toBe('button');
    expect(resetButton.text()).toBe('filter.date.clear');
  });
});

describe('Add Events', () => {
  it('Add a date in from input', () => {
    const pickerElements = getDatePickerElements('.input-from-to--from');
    const input = pickerElements.rangeDayPicker.find('input');
    input.simulate('change', { target: { value: '30/05/20' } });
    expect(RangeDayPickerComponent.state().from.toString()).toBe('Sat May 30 2020 00:00:00 GMT+0200 (GMT+02:00)');
    expect(defaultProps.dateFrom.mock.calls[0][0].toString()).toBe('Sat May 30 2020 00:00:00 GMT+0200 (GMT+02:00)');
  });
  it('Add a date in to input ', () => {
    const pickerElementsTo = getDatePickerElements('.input-from-to--to');
    const inputTo = pickerElementsTo.rangeDayPicker.find('input');
    inputTo.simulate('change', { target: { value: '30/05/20' } });
    expect(defaultProps.dateTo.mock.calls[0][0].toString()).toBe('Sat May 30 2020 00:00:00 GMT+0200 (GMT+02:00)');
  });
  it('Filled from and to inputs', () => {
    const pickerElementsTo = getDatePickerElements('.input-from-to--to');
    const pickerElements = getDatePickerElements('.input-from-to--from');
    const inputTo = pickerElementsTo.rangeDayPicker.find('input');
    const inputFrom = pickerElements.rangeDayPicker.find('input');
    inputFrom.simulate('change', { target: { value: '29/05/20' } });
    inputTo.simulate('change', { target: { value: '30/05/20' } });
    expect(defaultProps.dateFrom.mock.calls[0][0].toString()).toBe('Fri May 29 2020 00:00:00 GMT+0200 (GMT+02:00)');
    expect(defaultProps.dateTo.mock.calls[0][0].toString()).toBe('Sat May 30 2020 00:00:00 GMT+0200 (GMT+02:00)');
    expect(RangeDayPickerComponentInstance.showFromMonth.mock.calls).toHaveLength(1);
  });

  it('Click in reset dates ', () => {
    RangeDayPickerComponentInstance.setState = jest.fn();
    RangeDayPickerComponentInstance.toRef.current.setState = jest.fn();
    RangeDayPickerComponentInstance.fromRef.current.setState = jest.fn();
    const resetButton = RangeDayPickerComponent.find('[data-action="reset"]');
    resetButton.simulate('click');
    expect(defaultProps.resetDate.mock.calls[0][0]).toBeUndefined();
    expect(defaultProps.resetDate.mock.calls[0][1]).toBeUndefined();
    expect(RangeDayPickerComponentInstance.setState.mock.calls[0][0]).toEqual({ from: undefined, to: undefined });
    expect(RangeDayPickerComponentInstance.fromRef.current.setState.mock.calls[0][0]).toEqual({
      value: '',
      typedValue: '',
    });
    expect(RangeDayPickerComponentInstance.toRef.current.setState.mock.calls[0][0]).toEqual({
      value: '',
      typedValue: '',
    });
  });
});
