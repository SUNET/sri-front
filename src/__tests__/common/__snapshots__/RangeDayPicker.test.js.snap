// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeDayPicker Component Renders correctly 1`] = `
<CreateRangeDayPickerWrapper>
  <div>
    <RangeDayPicker
      dateFrom={[MockFunction]}
      dateTo={[MockFunction]}
      resetDate={[MockFunction]}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "filter.daypicker.from",
            ],
            Array [
              "filter.daypicker.to",
            ],
            Array [
              "filter.date.clear",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "filter.daypicker.from",
            },
            Object {
              "type": "return",
              "value": "filter.daypicker.to",
            },
            Object {
              "type": "return",
              "value": "filter.date.clear",
            },
          ],
        }
      }
    >
      <div
        className="range-day-picker"
      >
        <div
          className="input-from-to input-from-to--from"
        >
          <label>
            filter.daypicker.from
          </label>
          <DayPickerInput
            classNames={
              Object {
                "container": "DayPickerInput",
                "overlay": "DayPickerInput-Overlay",
                "overlayWrapper": "DayPickerInput-OverlayWrapper",
              }
            }
            clickUnselectsDay={false}
            component="input"
            dayPickerProps={
              Object {
                "disabledDays": Object {
                  "after": undefined,
                },
                "locale": "en",
                "localeUtils": Object {
                  "formatDate": [Function],
                  "formatDay": [Function],
                  "formatMonthTitle": [Function],
                  "formatWeekdayLong": [Function],
                  "formatWeekdayShort": [Function],
                  "getFirstDayOfWeek": [Function],
                  "getMonths": [Function],
                  "parseDate": [Function],
                },
                "modifiers": Object {
                  "end": undefined,
                  "start": undefined,
                },
                "numberOfMonths": 2,
                "onDayClick": [Function],
                "selectedDays": Array [
                  undefined,
                  Object {
                    "from": undefined,
                    "to": undefined,
                  },
                ],
                "toMonth": undefined,
              }
            }
            format="DD/MM/YY"
            formatDate={[Function]}
            hideOnDayClick={true}
            inputProps={Object {}}
            keepFocus={true}
            onDayChange={[Function]}
            overlayComponent={[Function]}
            parseDate={[Function]}
            placeholder="dd/mm/yy"
            showOverlay={false}
            typedValue=""
            value=""
          >
            <div
              className="DayPickerInput"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                placeholder="dd/mm/yy"
                value=""
              />
            </div>
          </DayPickerInput>
        </div>
        <div
          className="input-from-to input-from-to--to"
        >
          <label>
            filter.daypicker.to
          </label>
          <DayPickerInput
            classNames={
              Object {
                "container": "DayPickerInput",
                "overlay": "DayPickerInput-Overlay",
                "overlayWrapper": "DayPickerInput-OverlayWrapper",
              }
            }
            clickUnselectsDay={false}
            component="input"
            dayPickerProps={
              Object {
                "disabledDays": Object {
                  "before": undefined,
                },
                "fromMonth": undefined,
                "locale": "en",
                "localeUtils": Object {
                  "formatDate": [Function],
                  "formatDay": [Function],
                  "formatMonthTitle": [Function],
                  "formatWeekdayLong": [Function],
                  "formatWeekdayShort": [Function],
                  "getFirstDayOfWeek": [Function],
                  "getMonths": [Function],
                  "parseDate": [Function],
                },
                "modifiers": Object {
                  "end": undefined,
                  "start": undefined,
                },
                "month": undefined,
                "numberOfMonths": 2,
                "selectedDays": Array [
                  undefined,
                  Object {
                    "from": undefined,
                    "to": undefined,
                  },
                ],
              }
            }
            format="DD/MM/YY"
            formatDate={[Function]}
            hideOnDayClick={true}
            inputProps={Object {}}
            keepFocus={true}
            onDayChange={[Function]}
            overlayComponent={[Function]}
            parseDate={[Function]}
            placeholder="dd/mm/yy"
            showOverlay={false}
            typedValue=""
            value=""
          >
            <div
              className="DayPickerInput"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                placeholder="dd/mm/yy"
                value=""
              />
            </div>
          </DayPickerInput>
        </div>
        <button
          className="range-day-picker__reset_dates btn outline btn-remove"
          data-action="reset"
          onClick={[Function]}
          type="button"
        >
          filter.date.clear
        </button>
      </div>
    </RangeDayPicker>
  </div>
</CreateRangeDayPickerWrapper>
`;
